staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation.go

```diff
@@ -141,7 +159,12 @@ func (s *Validator) validateExpressions(fldPath *field.Path, sts *schema.Structu
 			continue // todo: wire oldObj parameter
 		}
 		evalResult, evalDetails, err := compiled.Program.Eval(activation)
-
+		rtCost := evalDetails.ActualCost()
+		if *rtCost > math.MaxInt || int64(*rtCost) > remainingBudget {
```

Comment:
I was using nagative raminingBuget to fail the validation. It doesn't save much by switching to uint

Reply:
I'd comment this carefully and explain how this prevents underflow.  I realize that `remainingBudget` is never expected to be < 0 here but it might be worth validating since the underflow prevention assumes that. The alternative would be to do something like `if *rtCost >= remainingBudget { remainingBudget == 0 } else { remainingBudget -= *rtCost}` and use uint64 (and 0 to represent an exhausted  budget). Both work, uint64 does seem simpler to me but I'll let you decide.